CMakeLists.txt

```
@@ -34,11 +34,20 @@ list ( APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/" )
 # Find Julia
 find_package( Julia REQUIRED )
 
-
-
 # Find MPI
 find_package( MPI REQUIRED )
 
+# Find google test on demand
+option( TESTING "Build tests using google test" )
+if( TESTING )
```

Is `TESTING` a canonical keyword here? Otherwise, I'd suggest to use something more explicit such as `ENABLE_TESTS` or `ENABLE_TESTING` etc.

Also:
```suggestion
option( TESTING "Build tests using Google Test" )
if( TESTING )
```

---

There was no specific reason for TESTING. ENABLE_TESTING is better!